plugins/module_utils/network/sonic/config/static_routes/static_routes.py

```diff
@@ -0,0 +1,341 @@
+        tag = next_hop.get('tag', None)
+        idx = self.generate_index(index)
+        if idx:
+            next_hop_cfg['index'] = idx
+            if blackhole:
+                next_hop_cfg['blackhole'] = blackhole
+            if network_instance:
+                next_hop_cfg['network-instance'] = network_instance
+            if next_hop:
+                next_hop_cfg['next-hop'] = next_hop_attr
+            if metric:
+                next_hop_cfg['metric'] = metric
+            if track:
+                next_hop_cfg['track'] = track
+            if tag:
+                next_hop_cfg['tag'] = tag
+
+        url = '%s=%s/%s' % (network_instance_path, vrf_name, protocol_static_routes_path)
+        next_hops_cfg = {'next-hop': [{'index': idx, 'config': next_hop_cfg}]}
+        if interface:
+            next_hops_cfg['next-hop'][0]['interface-ref'] = {'config': {'interface': interface}}
+        payload = {'openconfig-network-instance:static-routes': {'static': [{'prefix': prefix, 'config': {'prefix': prefix}, 'next-hops': next_hops_cfg}]}}
+        request = {'path': url, 'method': PATCH, 'data': payload}
+
+        return request
+
+    def generate_index(self, index):
+        idx = None
+        blackhole = index.get('blackhole', None)
+        interface = index.get('interface', None)
+        network_instance = index.get('network_instance', None)
+        next_hop = index.get('next_hop', None)
+
+        if blackhole is True:
+            idx = 'DROP'
+        else:
+            if interface and not next_hop and not network_instance:
+                idx = interface
+            if interface and next_hop and not network_instance:
+                idx = interface + '_' + next_hop
+            if interface and network_instance and not next_hop:
+                idx = interface + '_' + network_instance
+            if interface and next_hop and network_instance:
+                idx = interface + '_' + next_hop + '_' + network_instance
+            if next_hop and not interface and not network_instance:
+                idx = next_hop
+            if next_hop and network_instance and not interface:
+                idx = next_hop + '_' + network_instance
+
+        return idx
+
+    def get_delete_static_routes_requests(self, commands, have, is_delete_all):
+        requests = []
+        if is_delete_all:
+            requests = self.get_delete_all_static_routes_requests(commands)
+        else:
+            for cmd in commands:
+                vrf_name = cmd.get('vrf_name', None)
+                static_list = cmd.get('static_list', [])
+                for static in static_list:
+                    prefix = static.get('prefix', None)
+                    next_hops = static.get('next_hops', [])
+
+                    for cfg in have:
+                        cfg_vrf_name = cfg.get('vrf_name', None)
+                        if vrf_name == cfg_vrf_name:
+                            cfg_static_list = cfg.get('static_list', [])
```

Comment:
Can the user delete all prefixes for a VRF by specifying a VRF with no prefix list?

If so, handling for that case needs to be added.